### ISSUE TITLE
Persist sidebar state (categories, hidden)

### ISSUE THREAD

**Issue by Josh-Cena** [facebook/docusaurus#6319]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

It would be useful to persist sidebar state like which categories are collapsed and if the sidebar is hidden or not in local storage (or session storage?), so that the user's navigation experience is not interrupted by, say, going to the blog briefly.

### Has this been requested on Canny?

No

### Motivation

Better UX

### API design

Persist state in local storage / session storage and retrieve on sidebar first render. After #3811 we probably have a better way to control the collapsed state of individual categories.

### Have you tried building it?

No

### Self-service

- [ ] I'd be willing to contribute this feature to Docusaurus myself.

**Comment by erikvanoosten** [comment]
On https://zio.dev/ecosystem/ the left-side navigation is pretty large. It would be nice if it automatically scrolls to the active page at load time. Is that part of this issue?

**Comment by Josh-Cena** [comment]
@erikvanoosten No, but unfortunately we had troubles with scroll-to-active features in the past: https://github.com/facebook/docusaurus/pull/6317, https://github.com/facebook/docusaurus/pull/6748, https://github.com/facebook/docusaurus/pull/6666 so it won't be soon that we introduce this feature to left sidebars.
